boa_engine/src/syntax/parser/expression/update.rs

```diff
@@ -57,56 +57,66 @@ where
         let _timer = Profiler::global().start_event("UpdateExpression", "Parsing");
 
         let tok = cursor.peek(0, interner)?.ok_or(ParseError::AbruptEnd)?;
+        let position = tok.span().start();
         match tok.kind() {
             TokenKind::Punctuator(Punctuator::Inc) => {
                 cursor
                     .next(interner)?
                     .expect("Punctuator::Inc token disappeared");
-                return Ok(node::UnaryOp::new(
-                    UnaryOp::IncrementPre,
-                    UnaryExpression::new(self.name, self.allow_yield, self.allow_await)
-                        .parse(cursor, interner)?,
-                )
-                .into());
+
+                let target = UnaryExpression::new(self.name, self.allow_yield, self.allow_await)
+                    .parse(cursor, interner)?;
+
+                if cursor.strict_mode() {
+                    if let Node::Identifier(ident) = target {
+                        if ident.sym() == Sym::ARGUMENTS {
+                            return Err(ParseError::lex(LexError::Syntax(
+                                "unexpected identifier 'arguments' in strict mode".into(),
+                                position,
+                            )));
+                        } else if ident.sym() == Sym::EVAL {
+                            return Err(ParseError::lex(LexError::Syntax(
+                                "unexpected identifier 'eval' in strict mode".into(),
+                                position,
+                            )));
+                        }
+                    }
+                }
```

Comment:
This exact code is repeated twice. I think this should be a method in `ident`

Reply:
Fixed & rebased